alphagov/paas-cf
# Introduce custom error page

What
----

We'd like to be able to display a custom error message whenever gorouter
has an issue, like a 404 page for instance.

This is certainly possible, however it only accepts a Go template.

Our custom page, should be using the govuk-frontend node module, mainly
for styles.

We'd like to make sure that static page is easily upgradable and
maintainable, which is the reason of throwing in some NodeJS and React
into the mix. TypeScript is a must, Rafal insists on having.

The application should be used to upgrade the YAML patch upon upgrades
or changes.

How to review
-------------

- deploy to your dev env (or see tlwr whilst it lasts)
- visit `https://404.${SYSTEM_DOMAIN}/`
- experience GOV.UK branded `Not Found` page
- optionally, attempt to run the application (see README.md)
- validate the tests are enough check
